frontend/packages/dev-console/src/components/formik-fields/field-types.ts

```diff
@@ -1,30 +1,53 @@
-export interface InputFieldProps {
+import { TextInputTypes } from '@patternfly/react-core';
+
+export interface FieldProps {
   name: string;
-  type?: string;
   label?: string;
   helpText?: string;
   required?: boolean;
+}
+
+export interface InputFieldProps extends FieldProps {
+  type: TextInputTypes;
   placeholder?: string;
   onChange?: (event) => void;
   onBlur?: (event) => void;
 }
 
+export interface CheckboxFieldProps extends FieldProps {
+  formLabel?: string;
+}
+
 export interface SearchInputFieldProps extends InputFieldProps {
   onSearch: (searchTerm: string) => void;
 }
 
-export interface DropdownFieldProps extends InputFieldProps {
+export interface DropdownFieldProps extends FieldProps {
   items?: object;
   selectedKey?: string;
   title?: React.ReactNode;
   fullWidth?: boolean;
 }
 
-export interface EnvironmentFieldProps extends InputFieldProps {
+export interface EnvironmentFieldProps extends FieldProps {
   obj?: object;
   envPath: string[];
 }
 
+export interface ResourceLimitFieldProps extends FieldProps {
```

Comment:
Why did you put Resource Limit in the formik types. Just think that it should be in the components not in the things that make forms.

Reply:
ok, so I see that you didn't put it here. My concern is still there, why is this in the forms and not the compenents section?

Reply:
Because this is just wrapping `RequestSizeInput` component in a `FormGroup` and connecting to formik context. So, its a formik field which any form can use. Its similar to `EnvironmentField`, `DroppableFileInputField` and `NumberSpinnerField`.